src/builtin.c

```diff
@@ -1186,9 +1186,20 @@ static jv f_string_indexes(jq_state *jq, jv a, jv b) {
 }
 
 static jv f_string_implode(jq_state *jq, jv a) {
+  int i;
+  int len;
   if (jv_get_kind(a) != JV_KIND_ARRAY) {
     return ret_error(a, jv_string("implode input must be an array"));
   }
+  len = jv_array_length(jv_copy(a));
+  for (i = 0; i < len; i++) {
+    jv n = jv_array_get(jv_copy(a), i);
+    if (jv_get_kind(n) != JV_KIND_NUMBER) {
```

Comment:
From reading the code it seems like there is not much error handling and mostly asserts in utils.c where jv_string_implode is so that is why i added error checks here